File: Cognito.Client/src/app/data/schema/domain.ts
import { Tenant } from './tenant';
import { DomainLicense } from './domain-license';
import { UserDomain } from './user-domain';

export class Domain {
    id: number;
    name: string;
    basicLicenses: number;
    timeKeeperLicenses: number;
    companyInformation: string;
    adminFirstName: string;
    adminLastName: string;
    adminEmail: string;
    tenant: Tenant;
    domainLicenses: DomainLicense[];
    userDomains: UserDomain[];
}

File: Cognito.Server/Cognito.DataAccess/Entities/Domain.cs
using Cognito.DataAccess.DbContext.Abstract;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace Cognito.DataAccess.Entities
{
    public class Domain : AuditableEntityBase
    {
        [Required]
        [MaxLength(128)]
        public string Name { get; set; }

        public int? AdminUserId { get; set; }

        public int? TenantId { get; set; }

        public virtual Tenant Tenant { get; set; }

        public virtual ICollection<UserDomain> UserDomains { get; set; }

        public virtual ICollection<Project> Projects { get; set; }

        public virtual ICollection<Contact> Contacts { get; set; }

        public virtual ICollection<DomainLicense> DomainLicenses { get; set; }
    }
}
